service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java

```diff
@@ -71,6 +73,33 @@ private static String getRoleSessionName(String value) {
         : value;
   }
 
+  public static void appendUserTags(Collection<Tag> tags, AuthenticatedUserRequest userRequest) {
+    tags.add(Tag.builder().key("user_id").value(userRequest.getSubjectId()).build());
+  }
+
+  public static void appendResourceTags(Collection<Tag> tags, AwsCloudContext awsCloudContext) {
+    tags.add(Tag.builder().key("version").value(awsCloudContext.getMajorVersion()).build());
+    tags.add(Tag.builder().key("tenant").value(awsCloudContext.getTenantAlias()).build());
+    tags.add(Tag.builder().key("environment").value(awsCloudContext.getEnvironmentAlias()).build());
+  }
+
+  public static void appendPrincipalTags(
+      Collection<Tag> tags,
+      AwsCloudContext awsCloudContext,
+      ControlledAwsStorageFolderResource awsStorageFolderResource) {
+    tags.add(Tag.builder().key("version").value(awsCloudContext.getMajorVersion()).build());
+    tags.add(
+        Tag.builder().key("bucket_id").value(awsStorageFolderResource.getBucketName()).build());
+    tags.add(
+        Tag.builder().key("terra_bucket_id").value(awsStorageFolderResource.getPrefix()).build());
+  }
+
+  public static void appendRoleTags(Collection<Tag> tags, ApiAwsCredentialAccessScope accessScope) {
+    if (accessScope == ApiAwsCredentialAccessScope.WRITE_READ) {
+      tags.add(Tag.builder().key("ws_role").value("writer").build());
```

Comment:
`ws_role -> WorkspaceRole` per above comment.

Also, `WorkspaceRole` tag must be set to `reader` if the role is `READ_ONLY` (i.e. if not `WRITE_READ`).

Reply:
I read this as "will only be mentioned _**in the document**_ when it is writer" (and this is the case in the subsequent sections describing the IAM roles).  @jwtracy can confirm.

The sentence before that states "For the TerraUser role, this tag must always exist.", and reading the IAM policies aligns to this

Reply:
noted